sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/SchedulerBuilder.java

```diff
@@ -215,6 +217,40 @@ public SchedulerBuilder setPlanCustomizer(PlanCustomizer planCustomizer) {
         return this;
     }
 
+    public SchedulerBuilder withSingleRegionConstraint() {
+         Optional<String> schedulerRegion = schedulerConfig.getSchedulerRegion();
+         PlacementRule regionRule = getRegionRule(schedulerRegion);
+
+         List<PodSpec> updatedPodSpecs = serviceSpec.getPods().stream()
+                 .map(p -> podWithPlacementRule(p, regionRule))
+                 .collect(Collectors.toList());
+
+         DefaultServiceSpec.Builder builder = DefaultServiceSpec.newBuilder(serviceSpec).pods(updatedPodSpecs);
+         if (schedulerRegion.isPresent()) {
+             builder.region(schedulerRegion.get());
+         }
+         serviceSpec = builder.build();
+
+         return this;
+    }
+
+    @VisibleForTesting
+    static PlacementRule getRegionRule(Optional<String> schedulerRegion) {
+        if (!schedulerRegion.isPresent()) {
+            return new IsLocalRegionRule();
+        }
+
+        return RegionRuleFactory.getInstance().require(ExactMatcher.create(schedulerRegion.get()));
```

Comment:
Would it not make sense to have the `RegionRuleFactory` contain the logic for constructing the `IsLocalRegionRule` too? That would leave us with a *single* point for defining region -> region rule Mappings.

Reply:
Potentially -- those classes already existed as is, but I intend to do a refactor pass where I could adjust that.